util/plugins/eu.esdihumboldt.util/src/eu/esdihumboldt/util/geometry/WindingOrder.java

```diff
@@ -0,0 +1,205 @@
+
+		// Get coordinates of geometry
+		Coordinate[] coordinates = geometry.getCoordinates();
+		// Using algorithm get order of geometry
+		boolean orientation = false;
+		try {
+			orientation = CGAlgorithms.isCCW(coordinates);
+		} catch (IllegalArgumentException ex) {
+			System.err.println("Illegal Argument");
+			// TODO:: or Throw ?
+		}
+
+		return orientation;
+
+	}
+
+	/**
+	 * Unify Winding order of Geometry object( including Polygon, Multipolygon,
+	 * LinearRing) as CounterClockwise or Clockwise.
+	 * 
+	 * @param geometry Geometry object for unifying.
+	 * @param counterClockWise
+	 * @return Geometry unified object.
+	 */
+	public static Geometry unifyWindingOrder(Geometry geometry, boolean counterClockWise) {
+
+		if (geometry == null) {
+			return null;
+		}
+
+		if (geometry instanceof MultiPolygon) {
+			return unifyWindingOrderForMultiPolygon((MultiPolygon) geometry, counterClockWise);
+		}
+		else if (geometry instanceof Polygon) {
+			return unifyWindingOrderForPolyGon((Polygon) geometry, counterClockWise);
+		}
+		else if (geometry instanceof LinearRing) {
+			return unifyWindingOrderForLinearRing((LinearRing) geometry, counterClockWise);
+		}
+		else if (geometry instanceof GeometryCollection) {
+			return unifyWindingOrderForGeometryCollection((GeometryCollection) geometry,
+					counterClockWise);
+		}
+		else
+			return geometry;
+	}
+
+	/**
+	 * Unify order for LinearRing as CounterClockwise or Clockwise.
+	 * 
+	 * @param linearRing LinearRing object for unifying
+	 * @param counterClockWise boolean value. true, if want geometry as counter
+	 *            clock wise, else false.
+	 * @return LinearRing unified object.
+	 */
+	public static LinearRing unifyWindingOrderForLinearRing(LinearRing linearRing,
+			boolean counterClockWise) {
+
+		// Checking and reversing geometry
+		if (isCounterClockwise(linearRing) == counterClockWise)
+			return linearRing;
+		else
+			return (LinearRing) linearRing.reverse();
+	}
+
+	/**
+	 * Unify order of the polygon as counterClockwise or not including all its
+	 * holes.
+	 * 
+	 * @param poly Polygon object for unifying
+	 * @param counterClockWise boolean value. true, if want shell of Polygon as
+	 *            counter clock wise and holes as clockwise, else false.
+	 * @return Polygon unified object.
+	 */
+	public static Polygon unifyWindingOrderForPolyGon(Polygon poly, boolean counterClockWise) {
+
+		// Checking and reversing Shell
+		LinearRing shell = unifyWindingOrderForLinearRing((LinearRing) poly.getExteriorRing(),
+				counterClockWise);
+
+		// Checking and reversing Holes
+		LinearRing holes[] = new LinearRing[poly.getNumInteriorRing()];
+		for (int i = 0; i < poly.getNumInteriorRing(); i++) {
+			holes[i] = unifyWindingOrderForLinearRing((LinearRing) poly.getInteriorRingN(i),
+					!counterClockWise);
+		}
+
+		// Create New Polygon using unified shell and holes
+		Polygon tempPoly = factory.createPolygon(shell,
+				poly.getNumInteriorRing() > 0 ? holes : null);
```

Comment:
If you do the check `poly.getNumInteriorRing() > 0` anyway, you could do it earlier and if the condition is met not create an array in the first place.